### PR TITLE
[ja] Translate tasks/debug-application-cluster/audit into Japanese

### LINKED ISSUE #30564: [ja] Translate tasks/debug-application-cluster/audit into Japanese
# This is a Feature Request

## What would you like to be added

Translate [tasks/debug-application-cluster/audit](https://kubernetes.io/docs/tasks/debug-application-cluster/audit)  into Japanese.

## Why is this needed

This page has not been translated.

## Comments

/language ja
/assign

### PR DESCRIPTION
**Pull request by ptux**
fix #30564

The following files are not targeted to translate.
They are required for reference.

- content/ja/docs/reference/command-line-tools-reference/kube-apiserver.md
- content/ja/docs/reference/config-api/apiserver-audit.v1.md

### CODE REVIEW

**Comment by ptux** [comment]
Ready for review.

**Comment by ptux** [comment]
https://github.com/kubernetes/website/issues/30779 is closed.

File: content/ja/docs/reference/command-line-tools-reference/kube-apiserver.md [hunk 1]
```diff
@@ -0,0 +1,2535 @@
+---
```
Comment (nasa9084) [inline content/ja/docs/reference/command-line-tools-reference/kube-apiserver.md:1]: This file seems written in Chinese, please translate into Japanese or copy from English

File: content/ja/docs/reference/config-api/apiserver-audit.v1.md [hunk 2]
```diff
@@ -0,0 +1,620 @@
+---
```
Comment (nasa9084) [inline content/ja/docs/reference/config-api/apiserver-audit.v1.md:1]: just copied from English, right?

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 3]
```diff
@@ -0,0 +1,238 @@
+---
+content_type: concept
+title: 監査
+---
+
+<!-- overview -->
+
+Kubernetesの監査はクラスタ内の一連の行動を記録するセキュリティに関連した時系列の記録を提供します。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:8]:
```suggestion
Kubernetesの監査はクラスター内の一連の行動を記録するセキュリティに関連した時系列の記録を提供します。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+クラスタはユーザー、Kubernetes APIを使用するアプリケーション、
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:9]:
```suggestion
クラスターはユーザー、Kubernetes APIを使用するアプリケーション、
```

Comment (nasa9084) [reply content/ja/docs/tasks/debug-application-cluster/audit.md:9]: 以下同様にお願いします

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+ - いつ起こったのか？
+ - 誰がそれを始めたのか？
+ - 何のために起こったのか？
+ - それはどこで観察されましたか？
+ - それはどこから始まったのか？
+ - それはどこへ向かっていたのか？
+
+<!-- body -->
+
+監査記録は、そのライフサイクルを[kube-apiserver](/docs/reference/command-line-tools-reference/kube-apiserver/)コンポーネントの中で始まります。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:24]:
```suggestion
監査記録のライフサイクルは[kube-apiserver](/docs/reference/command-line-tools-reference/kube-apiserver/)コンポーネントの中で始まります。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+各リクエストは、その実行の各段階でその実行の各段階で、監査イベントが生成されます。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:25]:
```suggestion
各リクエストの実行の各段階で、監査イベントが生成されます。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+ポリシーに従って前処理され、バックエンドに書き込まれます。 ポリシーが何を記録するかを決定しを決定し、
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:26]:
```suggestion
ポリシーに従って前処理され、バックエンドに書き込まれます。 ポリシーが何を記録するかを決定し、
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+バックエンドがその記録を永続化します。現在のバックエンドの実装はログファイルやWebhookなどがあります。
+
+各リクエストは関連する _stage_ で記録されます。
+定義されたステージは以下の通りです:
+
+- `RequestReceived` - 監査ハンドラーがリクエストを受信すると同時に生成されるイベントのステージ。
+  つまり、ハンドラーチェーンに委譲される前に生成されるイベントのステージです。
+- `ResponseStarted` - レスポンスヘッダーが送信された後、レスポンスボディが送信される前のステージです。
+  このステージは長時間実行されるリクエスト（watchなど）でのみ発生します。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:35]:
```suggestion
  このステージは長時間実行されるリクエスト(watchなど)でのみ発生します。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+- `ResponseComplete` - レスポンスボディの送信が完了して、それ以上のバイトは送信されません。
+- `Panic` - パニックが起きたときに発生するイベント。
+
+{{< note >}}
+[Audit Event configuration](/docs/reference/config-api/apiserver-audit.v1/#audit-k8s-io-v1-Event)の設定は[Event](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#event-v1-core)APIオブジェクトとは異なります。
+{{< /note >}}
+
+監査ログ機能は、リクエストごとに監査に必要なコンテキストが保存されるため、APIサーバーのメモリ消費量が増加します。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:43]:
```suggestion
監査ログ機能は、リクエストごとに監査に必要なコンテキストが保存されるため、APIサーバーのメモリー消費量が増加します。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+メモリの消費量は、監査ログ機能の設定によって異なります。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:44]:
```suggestion
メモリーの消費量は、監査ログ機能の設定によって異なります。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+
+## Audit policy
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:46]:
```suggestion
## 監査ポリシー
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 12]
```diff
@@ -0,0 +1,238 @@
...
+
+監査ポリシーはどのようなイベントを記録し、どのようなデータを含むべきかについてのルールを定義します。
+監査ポリシーのオブジェクト構造は、[audit.k8s.io` API group](/docs/reference/config-api/apiserver-audit.v1/#audit-k8s-io-v1-Policy)で定義されています。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:49]:
```suggestion
監査ポリシーのオブジェクト構造は、[`audit.k8s.io` API group](/docs/reference/config-api/apiserver-audit.v1/#audit-k8s-io-v1-Policy)で定義されています。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 13]
```diff
@@ -0,0 +1,238 @@
...
+
+イベントが処理されると、そのイベントは順番にルールのリストと比較されます。
+最初のマッチングルールは、イベントの監査レベルを設定します。
+
+定義されている監査レベルは:
+
+- `None` - ルールに一致するイベントを記録しません。
+- `Metadata` - lリクエストのメタデータ（リクエストしたユーザー、タイムスタンプ、リソース、動作など）を記録しますが、リクエストやレスポンスのボディは記録しません。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:57]:
```suggestion
- `Metadata` - リクエストのメタデータ(リクエストしたユーザー、タイムスタンプ、リソース、動作など)を記録しますが、リクエストやレスポンスのボディは記録しません。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 14]
```diff
@@ -0,0 +1,238 @@
...
+- `Request` - ログイベントのメタデータとリクエストボディは表示されますが、レスポンスボディは表示されません。
+  これは非リソースリクエストには適用されません。
+- `RequestResponse` - イベントのメタデータ、リクエストとレスポンスのボディを記録しますが、
+  非リソースリクエストには適用されません。
+
+You can pass a file with the policy to `kube-apiserver`
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:63]: I think you forgot removing this block

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 15]
```diff
@@ -0,0 +1,238 @@
...
+using the `--audit-policy-file` flag. If the flag is omitted, no events are logged.
+Note that the `rules` field __must__ be provided in the audit policy file.
+A policy with no (0) rules is treated as illegal.
+
+`audit-policy-file`フラグを使って、ポリシーを記述したファイルを `kube-apiserver`に渡すことができます。
+このフラグが省略された場合イベントは記録されません。
+監査ポリシーファイルでは、`rules`フィールドが必ず指定されることに注意してください。
+ルールがない(0)ポリシーは不当なものとして扱われます。
+
+以下は監査ポリシーファイルの例:
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:73]:
```suggestion
以下は監査ポリシーファイルの例です:
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 16]
```diff
@@ -0,0 +1,238 @@
...
+# Log all requests at the Metadata level.
+apiVersion: audit.k8s.io/v1
+kind: Policy
+rules:
+- level: Metadata
+```
+
+独自の監査プロファイルを作成する場合は、Google Container-Optimized OSの監査プロファイルを出発点として使用できます。
+監査ポリシーファイルを生成する[configure-helper.sh](https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/configure-helper.sh)スクリプトを確認することができます。
+スクリプトを直視することで、監査ポリシーファイルのほとんどを見ることができます。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:89]:
```suggestion
スクリプトを直接見ることで、監査ポリシーファイルのほとんどを見ることができます。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 17]
```diff
@@ -0,0 +1,238 @@
...
+ログバックエンドは監査イベントを[JSONlines](https://jsonlines.org/)形式のファイルに書き込みます。
+以下の `kube-apiserver` フラグを使ってログ監査バックエンドを設定できます。
+
+- `--audit-log-path` は、ログバックエンドが監査イベントを書き込む際に使用するログファイルのパスを指定します。
+  このフラグを指定しないと、ログバックエンドは無効になります。`-` は標準出力を意味します。
+- `--audit-log-maxage` は、古い監査ログファイルを保持する最大日数を定義します。
+- `audit-log-maxbackup`は、保持する監査ログファイルの最大数を定義します。
+- `--audit-log-maxsize` は、監査ログファイルがローテーションされるまでの最大サイズをメガバイト単位で定義します。
+
+クラスタのコントロールプレーンでkube-apiserverをPodとして動作させている場合は、監査記録が永久化されるように、ポリシーファイルとログファイルの場所に`hostPath`をマウントすることを忘れないでください。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:135]:
```suggestion
クラスターのコントロールプレーンでkube-apiserverをPodとして動作させている場合は、監査記録が永久化されるように、ポリシーファイルとログファイルの場所に`hostPath`をマウントすることを忘れないでください。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 18]
```diff
@@ -0,0 +1,238 @@
...
+    type: File
+
+- name: audit-log
+  hostPath:
+    path: /var/log/audit.log
+    type: FileOrCreate
+
+```
+
+### Webhook バックエンド
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:169]:
```suggestion
### Webhookバックエンド
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 19]
```diff
@@ -0,0 +1,238 @@
...
+これは認証手段を含むKubernetes APIの形式であると想定されます。
+
+Webhook監査バックエンドを設定するには、以下のkube-apiserverフラグを使用します。
+
+- `--audit-webhook-config-file` は、Webhookの設定ファイルのパスを指定します。
+  webhookの設定は、事実上特化した [kubeconfig](/docs/tasks/access-application-cluster/configure-access-multiple-clusters) です。
+- `--audit-webhook-initial-backoff` は、最初に失敗したリクエストの後、再試行するまでに待つ時間を指定します。
+  それ以降のリクエストは、指数関数的なバックオフで再試行されます。
+
+Webhookの設定ファイルは、kubeconfig 形式でサービスのリモートアドレスと接続に使用する認証情報を指定します。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:181]:
```suggestion
Webhookの設定ファイルは、kubeconfig形式でサービスのリモートアドレスと接続に使用する認証情報を指定します。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 20]
```diff
@@ -0,0 +1,238 @@
...
+
+- `--audit-webhook-batch-buffer-size`は、バッチ処理を行う前にバッファリングするイベントの数を定義します。
+  入力イベントの割合がバッファをオーバーフローすると、イベントはドロップされます。
+- `--audit-webhook-batch-max-size`は、1つのバッチに入れるイベントの最大数を定義します。
+- `--audit-webhook-batch-max-wait`は、キュー内のイベントを無条件にバッチ処理するまでの最大待機時間を定義します。
+- `--audit-webhook-batch-throttle-qps`は、1秒あたりに生成されるバッチの最大平均数を定義します。
+- `--audit-webhook-batch-throttle-burst`は、許可された QPS が低い場合に、同じ瞬間に生成されるバッチの最大数を定義します。
+
+
+## パラメータチューニング
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:206]:
```suggestion
## パラメーターチューニング
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 21]
```diff
@@ -0,0 +1,238 @@
...
+
+パラメータは、APIサーバーの負荷に合わせて設定してください。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:208]:
```suggestion
パラメーターは、APIサーバーの負荷に合わせて設定してください。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 22]
```diff
@@ -0,0 +1,238 @@
...
+
+例えば、kube-apiserverが毎秒100件のリクエストを受け取り、それぞれのリクエストが`ResponseStarted`と`ResponseComplete`の段階でのみ監査されるとします。毎秒≅200の監査イベントが発生すると考えてください。
+1 つのバッチに最大 100 個のイベントがあるの場合、スロットリングレベルを少なくとも2クエリ/秒に設定する必要があります。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:211]:
```suggestion
1つのバッチに最大100個のイベントがあるの場合、スロットリングレベルを少なくとも2クエリ/秒に設定する必要があります。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 23]
```diff
@@ -0,0 +1,238 @@
...
+バックエンドがイベントを書き込むのに最大で5秒かかる場合、5秒分のイベントを保持するようにバッファサイズを設定する必要があります。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:212]:
```suggestion
バックエンドがイベントを書き込むのに最大で5秒かかる場合、5秒分のイベントを保持するようにバッファーサイズを設定する必要があります。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 24]
```diff
@@ -0,0 +1,238 @@
...
+
+logバックエンドとwebhookバックエンドは、ログに記録されるイベントのサイズを制限することをサポートしています。
+
+例として、logバックエンドで利用可能なフラグの一覧を以下に示します
+
+- `audit-log-truncate-enabled` イベントとバッチの切り捨てを有効にするかどうかです。
+- `audit-log-truncate-max-batch-size` バックエンドに送信されるバッチのバイト単位の最大サイズ。
+- `audit-log-truncate-max-event-size` バックエンドに送信される監査イベントのバイト単位の最大サイズです。
+
+デフォルトでは、`webhook`と`log`の両方で切り捨ては無効になっていますが、クラスタ管理者は `audit-log-truncate-enabled`または`audit-webhook-truncate-enabled`を設定して、この機能を有効にする必要があります。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:232]:
```suggestion
デフォルトでは、`webhook`と`log`の両方で切り捨ては無効になっていますが、クラスター管理者は `audit-log-truncate-enabled`または`audit-webhook-truncate-enabled`を設定して、この機能を有効にする必要があります。
```

**Review by nasa9084 — CHANGES_REQUESTED** [review]
後半はフォーマットのみ見てます

File: content/ja/docs/reference/config-api/apiserver-audit.v1.md (hunk 2 above)
Comment (ptux) [reply content/ja/docs/reference/config-api/apiserver-audit.v1.md:1]: Yes.

File: content/ja/docs/reference/command-line-tools-reference/kube-apiserver.md (hunk 1 above)
Comment (ptux) [reply content/ja/docs/reference/command-line-tools-reference/kube-apiserver.md:1]: changed to the English content.

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 25]
```diff
@@ -0,0 +1,235 @@
+---
+content_type: concept
+reviewers:
+- ptux
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:4]: please remove reviewers block

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 26]
```diff
@@ -0,0 +1,235 @@
...
+クラスターはユーザー、Kubernetes APIを使用するアプリケーション、
+およびコントロールプレーン自体によって生成されたアクティビティなどを監査します。
+
+監査により、クラスター管理者は以下の質問に答えることができます:
+
+ - 何が起きたのか？
+ - いつ起こったのか？
+ - 誰がそれを始めたのか？
+ - 何のために起こったのか？
+ - それはどこで観察されましたか？
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:20]:
```suggestion
 - それはどこで観察されたのか？
```

他の行と合わせましょう

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 27]
```diff
@@ -0,0 +1,235 @@
...
+
+### Webhookバックエンド
+
+Webhook監査バックエンドは、監査イベントをリモートのWeb APIに送信しますが、
+これは認証手段を含むKubernetes APIの形式であると想定されます。
+
+Webhook監査バックエンドを設定するには、以下のkube-apiserverフラグを使用します。
+
+- `--audit-webhook-config-file` は、Webhookの設定ファイルのパスを指定します。
+  webhookの設定は、事実上特化した [kubeconfig](/docs/tasks/access-application-cluster/configure-access-multiple-clusters) です。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:174]:
```suggestion
  webhookの設定は、事実上特化した[kubeconfig](/docs/tasks/access-application-cluster/configure-access-multiple-clusters)です。
```

**Review by nasa9084 — CHANGES_REQUESTED** [review]

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 28]
```diff
@@ -0,0 +1,235 @@
...
+例えば、kube-apiserverが毎秒100件のリクエストを受け取り、それぞれのリクエストが`ResponseStarted`と`ResponseComplete`の段階でのみ監査されるとします。毎秒≅200の監査イベントが発生すると考えてください。
+1つのバッチに最大100個のイベントがあるの場合、スロットリングレベルを少なくとも2クエリ/秒に設定する必要があります。
+バックエンドがイベントを書き込むのに最大で5秒かかる場合、5秒分のイベントを保持するようにバッファーサイズを設定する必要があります。
+
+10バッチ、または1000イベントとなります。
+
+しかし、ほとんどの場合デフォルトのパラメーターで十分であり、手動で設定する必要はありません。
+kube-apiserverが公開している以下のPrometheusメトリクスや、ログを見て監査サブシステムの状態を監視することができます。
+
+- `apiserver_audit_event_total` メトリックには、エクスポートされた監査イベントの合計数が含まれます。
```
Comment (riita10069) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:216]:
```suggestion
- `apiserver_audit_event_total`メトリックには、エクスポートされた監査イベントの合計数が含まれます。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 29]
```diff
@@ -0,0 +1,235 @@
...
+- `apiserver_audit_error_total` メトリックには、エクスポート中にエラーが発生してドロップされたイベントの総数が含まれます。
```
Comment (riita10069) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:217]:
```suggestion
- `apiserver_audit_error_total`メトリックには、エクスポート中にエラーが発生してドロップされたイベントの総数が含まれます。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 30]
```diff
@@ -0,0 +1,235 @@
...
+
+### ログエントリー・トランケーション {#truncate}
+
+logバックエンドとwebhookバックエンドは、ログに記録されるイベントのサイズを制限することをサポートしています。
+
+例として、logバックエンドで利用可能なフラグの一覧を以下に示します
+
+- `audit-log-truncate-enabled` イベントとバッチの切り捨てを有効にするかどうかです。
```
Comment (riita10069) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:225]:
```suggestion
- `audit-log-truncate-enabled`イベントとバッチの切り捨てを有効にするかどうかです。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 31]
```diff
@@ -0,0 +1,235 @@
...
+- `audit-log-truncate-max-batch-size` バックエンドに送信されるバッチのバイト単位の最大サイズ。
```
Comment (riita10069) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:226]:
```suggestion
- `audit-log-truncate-max-batch-size`バックエンドに送信されるバッチのバイト単位の最大サイズ。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md [hunk 32]
```diff
@@ -0,0 +1,235 @@
...
+- `audit-log-truncate-max-event-size` バックエンドに送信される監査イベントのバイト単位の最大サイズです。
```
Comment (riita10069) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:227]:
```suggestion
- `audit-log-truncate-max-event-size`バックエンドに送信される監査イベントのバイト単位の最大サイズです。
```

File: content/ja/docs/tasks/debug-application-cluster/audit.md (hunk 27 above)
Comment (riita10069) [inline content/ja/docs/tasks/debug-application-cluster/audit.md:174]:
```suggestion
  webhookの設定は、特殊な[kubeconfig](/ja/docs/tasks/access-application-cluster/configure-access-multiple-clusters)です。
```

**Pull request merged**